spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/ReflectUtils.scala

```diff
@@ -0,0 +1,112 @@
+    fields = fields ++ superFields
+
+    val values = new mutable.HashMap[String, AnyRef]()
+    fields.foreach(field => {
+      field.setAccessible(true)
+      values(field.getName) = field.get(o)
+    })
+    values
+  }
+
+  // create layer2 object form layer1
+  private def reflection(layer1: Object, layer2: Class[_],
+     tags: Array[ClassTag[_]], numerics: Array[TensorNumeric[_]]) : Object = {
+    val nameAndValues = getFieldNameAndValues(layer1)
+    val constructorMirror = getCostructorMirror(layer2)
+    val constructorFullParams = constructorMirror.symbol.paramss
+    val args = new Array[Object](constructorFullParams.map(_.size).sum)
+
+    val tagIter = tags.iterator
+    val numericIter = numerics.iterator
+    var i = 0
+    constructorFullParams.foreach(map => {
+      map.foreach(param => {
+        val name = param.name.decodedName.toString
+        val ptype = param.typeSignature
+        if (ptype <:< universe.typeOf[ClassTag[_]]||
+          ptype.typeSymbol == universe.typeOf[ClassTag[_]].typeSymbol) {
+          require(tagIter.hasNext, "If your module contains multiple class tags, " +
+            s"do you forget to override getClassTagNumerics method ${layer1}")
+          args(i) = tagIter.next()
+        } else if (ptype <:< universe.typeOf[TensorNumeric[_]]
+          || ptype.typeSymbol == universe.typeOf[TensorNumeric[_]].typeSymbol) {
+          args(i) = numericIter.next()
+        } else {
+          val value = nameAndValues.get(name).getOrElse(null)
+          args(i) = value
+        }
+        i += 1
+      })
+    })
+    constructorMirror.apply(args : _*).asInstanceOf[Object]
+  }
+
+  // create Module form IRElement
+  def reflectFromIR[T : ClassTag](layer: IRElement[T], cls: Class[_]) : Module[T] = {
+    val (tags, numerics) = layer.getOp().getClassTagNumerics()
+    val blasLayer = ReflectUtils.reflection(layer.getOp(), cls, tags, numerics)
+      .asInstanceOf[Module[T]]
+
+    if (blasLayer.parameters() != null) {
+      val params = blasLayer.getParameters()
+      val params2 = layer.getParameters()
+      if (params2._1 != null) params._1.copy(params2._1)
+      if (params2._2 != null) params._2.copy(params2._2)
+    }
+
+    if (layer.getName() != "") blasLayer.setName(layer.getName())
+
+    blasLayer
+  }
+
+  // create IRElement form Module
+  def reflectToIR[T: ClassTag](layer: Module[T], cls: Class[_]) : IRElement[T] = {
+    val (tags, numerics) = layer.getClassTagNumerics()
+    val op = ReflectUtils.reflection(layer, cls, tags, numerics).asInstanceOf[IROperator[T]]
+    val weightsAndBias =
+      if (layer.parameters() != null) layer.getParameters() else (null, null)
+    val element = IRElement[T](
+      layer.getName(), op, weights = weightsAndBias._1, gradWeights = weightsAndBias._2)
+    element
+  }
+
+  def findClass(name: String): Class[_] = {
+    try {
+      Class.forName(name)
+    } catch {
+      case ex: ClassNotFoundException => null
```

Comment:
why class not found should return but other exception will throw exception ?

Reply:
When finding class, generally there should be two results, class found and not found. But for other exceptions, I think it should throw exceptions directly.